libs/defi/shared/src/components/TxButton/hooks/useApprovalButton.tsx

```diff
@@ -103,104 +110,58 @@ export const useApprovalButton = (args: UseApprovalButtonProps) => {
   });
 
   const onWrite = useCallback(() => {
-    const act = {
-      type: 'approval',
-      title: intl.formatMessage(
-        { defaultMessage: 'Approve {token}' },
-        { token: args.token.symbol },
-      ),
-      subtitle: intl.formatMessage(
-        {
-          defaultMessage: 'Approving {token}',
-        },
-        {
-          token: args.token.symbol,
-        },
-      ),
-      endIcon: (
-        <BadgeIcon
-          badgeContent={
-            <FaCircleCheckRegular color="primary" sx={{ fontSize: 10 }} />
-          }
-        >
-          <TokenIcon token={args.token} sx={{ fontSize: 36 }} />
-        </BadgeIcon>
-      ),
+    const updated = pushActivity({
+      ...activity,
       status: 'pending',
-      tokenIn: args.token,
-      amountIn: args.amount,
-    } as const;
-    if (!args.disableActivity) {
-      const activity = pushActivity(act);
-      setAct(activity);
-    } else {
-      setAct({
-        id: Date.now().toString(),
-        createdOn: Date.now(),
-        ...act,
-      });
-    }
-    args.callbacks?.onWrite?.();
-  }, [
-    args.amount,
-    args.callbacks,
-    args.disableActivity,
-    args.token,
-    intl,
-    pushActivity,
-  ]);
+      title: activityOptions.approval.title(activity, intl),
+      subtitle: activityOptions.approval.subtitle(activity, intl),
+      icon: activityOptions.approval.icon(activity),
```

Comment:
This looks like it belongs on a `pushNotification` action but you have it here on `pushActivity`.

Reply:
the `activityOptions` are delivering the content, it is normal to feed it to any method that will later display it. Would it make it easier to understand if I'd split this file to another module?

Reply:
Title and subtitle are not valid inputs to activity, right?

Reply:
totally right, my bad, it's fixed